<launch>
    <param name="use_sim_time" value="false" />

    <node pkg="tf" type="static_transform_publisher" name="laser_publisher" args="0.32 0 0 0 0 0 base_link laser 10"/>
    <node pkg="tf" type="static_transform_publisher" name="imu_publisher" args="0.0 0 0 0 0 0 base_link imu_link 10"/>
    <node respawn="true" pkg="joy" type="joy_node" name="joystick">
        <param name="dev" type="string" value="/dev/input/js0"/>
        <param name="deadzone" value="0.05"/>
        <param name="autorepeat_rate" value="5"/>
        <param name="coalesce_interval" value="0.01"/>
    </node>

    <node name="imu_interface_broadcaster_node" pkg="imu_interface_broadcaster"
          type="imu_interface_broadcaster_node" output="screen"/>

    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
        <param name="base_footprint_frame" value="base_link"/>
        <param name="output_frame" value="odom"/>
        <param name="freq" value="30.0"/>
        <param name="sensor_timeout" value="1.0"/>
        <param name="odom_used" value="true"/>
        <param name="imu_used" value="true"/>
        <param name="vo_used" value="false"/>
        <param name="debug" value="false"/>
        <param name="self_diagnose" value="false"/>
        <remap from="odom" to="odom_raw"/>
        <remap from="imu_data" to="/imu/data_raw"/>
        <!--remap from="tf" to="tf_no_odom"/-->
    </node>
    <node pkg="rubber_navigation" type="odomFilter" name="odomFilter"/>
</launch>